Pod/Classes/AccessibilityHelpers/AccessibilityHelpers.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  AccessibilityHelpers.swift
+//  Pods
+//
+//  Created by Michael Skiba on 11/17/16.
+//
+//
+
+import UIKit
+
+public typealias AccessibilityAnnounceCompletion = (_ anncouncedString: String?, _ success: Bool?) -> ()
+
+/// A set of handy UIAccessibility helpers
+public class Accessibility: NSObject {
+
+    public static let shared = Accessibility()
+
+    public var announceCompletion: AccessibilityAnnounceCompletion?
+
+    public override init() {
+        super.init()
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(Accessibility.handleAnnounceDidFinish(_:)),
+                                               name: NSNotification.Name.UIAccessibilityAnnouncementDidFinish,
+                                               object: nil)
+    }
+
+    /**
+     Check if VoiceOver is currently running (UIAccessibilityIsVoiceOverRunning()).
+     */
+    public static var isVoiceOverRunning: Bool {
+        return UIAccessibilityIsVoiceOverRunning()
+    }
+
+    /**
+     Announce a message via VoiceOver with optional completion callback (UIAccessibilityAnnouncementNotification).
+
+     - parameter text:       The text to be spoken.
+     - parameter completion: A block to be invoked when the announcement has completed.
+     */
+    public func announce(_ text: String, completion: AccessibilityAnnounceCompletion? = nil) {
+        announceCompletion = completion
```

Comment:
As a compromise, we could call a pending completion before reassigning it. But we could als make a FIFO queue.